swimster-ui/src/hooks/useListingDetail.js

```diff
@@ -0,0 +1,62 @@
+import { useState, useEffect } from 'react'
+import { useAuthContext, isUserAuthenticated } from "../contexts/auth"
+import { useNavigate } from 'react-router-dom'
+import { useForm } from 'react-hook-form'
+import apiClient from "../services/apiClient"
+
+export const useListingDetail = listingId => {
+    const { user, initialized } = useAuthContext()
+    const { register, handleSubmit } = useForm()
+    const [isFetching, setIsFetching] = useState(false)
+    const [isProcessing, setIsProcessing] = useState(false)
```

Comment:
might be good to be more specific

`isSubmitProcessing`

Reply:
done!